lib/reesmat.gi

```diff
@@ -128,18 +128,10 @@ InstallMethod(IsZeroSimpleSemigroup, "for a Rees 0-matrix semigroup",
 function(R)
   local i;
 
-  for i in Columns(R) do 
-    if ForAll(Rows(R), j-> Matrix(R)[i][j]=0) then 
-      return false;
-    fi;
-  od;
-  
-  for i in Rows(R) do 
-    if ForAll(Columns(R), j-> Matrix(R)[j][i]=0) then 
-      return false;
-    fi;
-  od;
-  
+  if ForAny(Columns(R), j -> ForAll(Rows(R), i -> Matrix(R)[j][i] = 0))
+      or ForAny(Rows(R), i -> ForAll(Columns(R), j -> Matrix(R)[j][i] = 0)) then 
+    return false;
```

Comment:
Minor optimization: perhaps store `Matrix(R)` into a local variable.

Reply:
I'll make this change